espnet2/utils/plot_sinc_filters.py

```diff
@@ -0,0 +1,260 @@
+            plt.savefig(img_path, bbox_inches="tight")
+            print("Plotted %s" % img_path)
+
+            plt.clf()
+            plt.xlabel("kernel index")
+            plt.plot(x, kernel)
+            plt.plot(x, pre_kernel, "--", alpha=0.5)
+            img_name = "filter_kernel_both_%s.%s" % (str(i).zfill(2), args.filetype)
+            img_path = os.path.join(args.out_folder, img_name)
+            plt.savefig(img_path, bbox_inches="tight")
+            print("Plotted %s" % img_path)
+
+            y = np.zeros_like(x_f)
+            y[F_mins[i] : F_maxs[i]] = 1.0
+            plt.clf()
+            plt.plot(x_f, y)
+            img_name = "filter_freq_%s.%s" % (str(i).zfill(2), args.filetype)
+            img_path = os.path.join(args.out_folder, img_name)
+            plt.savefig(img_path, bbox_inches="tight")
+            print("Plotted %s" % img_path)
+
+            pre_y = np.zeros_like(x_f)
+            pre_y[pre_F_mins[i] : pre_F_maxs[i]] = 1.0
+            plt.clf()
+            plt.plot(x_f, y)
+            plt.plot(x_f, pre_y)
+            img_name = "filter_freq_both_%s.%s" % (str(i).zfill(2), args.filetype)
+            img_path = os.path.join(args.out_folder, img_name)
+            plt.savefig(img_path, bbox_inches="tight")
+            print("Plotted %s" % img_path)
+
+    plt.clf()
+    filters = [32, 71, 113, 126]
+    fig, axs = plt.subplots(2, 2, sharex=True, sharey="row")
+
+    axs[0, 0].plot(x, kernels[filters[0]][0])
+    axs[0, 0].plot(x, pre_kernels[filters[0]][0], "--", alpha=0.5)
+    axs[0, 1].plot(x, kernels[filters[1]][0])
+    axs[0, 1].plot(x, pre_kernels[filters[1]][0], "--", alpha=0.5)
+    axs[1, 0].plot(x, kernels[filters[2]][0])
+    axs[1, 0].plot(x, pre_kernels[filters[2]][0], "--", alpha=0.5)
+    axs[1, 1].plot(x, kernels[filters[3]][0])
+    axs[1, 1].plot(x, pre_kernels[filters[3]][0], "--", alpha=0.5)
+
+    img_name = "filter_kernel_ensemble2.%s" % (args.filetype)
+    img_path = os.path.join(args.out_folder, img_name)
+    plt.savefig(img_path, bbox_inches="tight")
+    print("Plotted %s" % img_path)
+
+
+def main(argv):
+    """Load the model, generate kernel and bandpass plots."""
+    parser = get_parser()
+    args = parser.parse_args(argv)
+    model_path = args.model_path
+    sample_rate = args.sample_rate
+
+    model = torch.load(model_path, map_location="cpu")
+    if "model" in model:  # snapshots vs. model.acc.best
+        model = model["model"]
+    filters = model["preencoder.filters.f"].detach().cpu().numpy()
+    f_mins = np.abs(filters[:, 0])
+    f_maxs = np.abs(filters[:, 0]) + np.abs(filters[:, 1] - filters[:, 0])
+    F_mins, F_maxs = f_mins * sample_rate, f_maxs * sample_rate
+    F_mins, F_maxs = np.round(F_mins).astype(np.int), np.round(F_maxs).astype(np.int)
+
+    plot_filter_kernels(filters, sample_rate, args)
+
+    def plot_filters(indices, filename):
+        x = np.linspace(0, np.max(F_maxs), np.max(F_maxs) + 1)
```

Comment:
Please do not use internal function. 

Basically,  internal function shouldn't be used if it's not necessary

Reply:
I refactored the plotting functions.
There are then two nested functions `mel()` and `hz()` left. They depend on how parameter initialization is implemented. It is just easier to keep them nested instead of programming different versions.